### ISSUE TITLE
Smart commit does not work for the session’s first PostgreSQL query (SELECT will start an uncommitted transaction)

### ISSUE THREAD

**Issue by maffe** [dbeaver/dbeaver#20459]
### Description

The first `SELECT` PostgreSQL query always starts a transaction and does not commit it, although Auto-commit is enabled. Smart commit is enabled, too.

### DBeaver Version

Community Edition 23.1.1

### Operating System

Windows 10

### Database and driver

Database: PostgreSQL 15.3 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit
Driver: PostgreSQL JDBC Driver 42.2.11

Sybase ASE 12.0 with jTDS driver seems to work as intended, it shows `Auto` after starting DBeaver. ![grafik](https://github.com/dbeaver/dbeaver/assets/515302/579d407d-7ec0-48cc-bc39-e718afb02f8b)

### Steps to reproduce

1. Start DBeaver, DBeaver transaction indicator shows `None` ![grafik](https://github.com/dbeaver/dbeaver/assets/515302/84b62092-820c-471b-90d5-51d0762bbe27),
Auto-commit is not yet enabled (but should be) 
![grafik](https://github.com/dbeaver/dbeaver/assets/515302/b77577e8-20e8-4b67-bcf8-8f7d93637034)
2. Execute `SELECT 1` on a PostgreSQL server, transaction indicator shows 1 modifying statement ![grafik](https://github.com/dbeaver/dbeaver/assets/515302/eac11be0-d335-4d11-9140-684493878e1c),
Auto-commit is still disabled ![grafik](https://github.com/dbeaver/dbeaver/assets/515302/b77577e8-20e8-4b67-bcf8-8f7d93637034)
3. Rollback, DBeaver shows `Auto` mode ![grafik](https://github.com/dbeaver/dbeaver/assets/515302/579d407d-7ec0-48cc-bc39-e718afb02f8b),
Auto-commit gets enabled
![grafik](https://github.com/dbeaver/dbeaver/assets/515302/94bad8b8-9bfe-4b1b-8905-64c38ee12bd7)
4. Execute any amount of `SELECT` queries. There will be no uncommitted transaction. This is the desired behaviour and should happen even on the first `SELECT` query.

Enabling Auto-commit manually after startup is possible and DBeaver will show `Auto` ![grafik](https://github.com/dbeaver/dbeaver/assets/515302/579d407d-7ec0-48cc-bc39-e718afb02f8b),
but after closing DBeaver and restarting it Auto-commit will be disabled again.

### Additional context

Connection type is set to Production (also for the Sybase ASE connection for which Smart commit works correctly).

Connection settings (same for PostgreSQL and Sybase ASE):
![grafik](https://github.com/dbeaver/dbeaver/assets/515302/cea9d27d-1bb9-4a2e-851b-7f9027813272)

**Comment by LonwoLonwo** [comment]
Hello @maffe 

Could you please provide a screen from Preferences -> Connections -> Connection Types with the Production mode enabled?

Does the auto-commit settings is on for the production type in your case on this page and in the connection settings -> Initialization?

**Comment by maffe** [comment]
![grafik](https://github.com/dbeaver/dbeaver/assets/515302/4d580f68-3c8d-4242-b11a-eaad7664c2ae)
![grafik](https://github.com/dbeaver/dbeaver/assets/515302/a66b5c05-7d13-4cee-9a2e-03cde2f36977)
I think these are the default settings.

**Comment by LonwoLonwo** [comment]
So... I'm confused.
You want to have the "auto-commit mode", but you can't save this setting? Because on your screens, auto-commit is off.

**Comment by maffe** [comment]
I want to have `SELECT` queries automatically committed, but not `INSERT`/`UPDATE`/`DELETE`. That’s why I enabled smart commit mode. But the first query with PostgreSQL always uses manual commit mode, even if it's a `SELECT` query. Smart commit works for later queries, but it’s annoying having to commit/rollback the first query when it’s only `SELECT`.

I noticed this affects not only the first query after opening DBeaver, but each new query tab (press F3, select `New script`, enter and execute `SELECT 1` and the transaction for this tab will also not be committed).
